integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/LogicalPlanSerializer.java

```diff
@@ -5,61 +5,118 @@
+ * cannot be replaced with dynamic proxy classes. The major limitation of this approach is that we
+ * cannot use JsonTypeInfo annotation to enrich jsons with class names within the plan.
  */
 @Slf4j
 class LogicalPlanSerializer {
   private static final int MAX_SERIALIZED_PLAN_LENGTH =
       50000; // 50K UTF-8 chars should be ~200KB + some extra bytes added during json encoding
-  private final ObjectMapper mapper;
+  private final Object objectMapper;
+
+  /** relocate plugin rewrites by default all occurences of com.fasterxml.jackson */
+  private static final String UNSHADED_JACKSON_PACKAGE = "com.".trim() + "fasterxml.jackson";
 
   public LogicalPlanSerializer() {
-    mapper = new ObjectMapper();
+    // we need to use non-relocated Spark's ObjectMapper to serialize logical plan
+    objectMapper = getObjectMapper();
+
     try {
-      mapper.registerModule(
-          (Module)
-              Class.forName("com.fasterxml.jackson.module.scala.DefaultScalaModule$")
-                  .getDeclaredField("MODULE$")
-                  .get(null));
-    } catch (Exception t) {
-      log.warn("Can't register jackson scala module for serializing LogicalPlan");
+      Class clazz =
+          Class.forName(
+              UNSHADED_JACKSON_PACKAGE + ".databind.introspect.ClassIntrospector$MixInResolver");
+
+      /**
+       * Translation class which allows merging non-shaded Jackson available within Spark context
+       * with LogicalPlanMixinResolver that extends shaded Jackson MixinResolver from the shaded
+       * library.
+       */
+      Object resolver =
+          Proxy.newProxyInstance(
+              clazz.getClassLoader(),
+              new Class[] {clazz},
+              new InvocationHandler() {
+                LogicalPlanMixinResolver resolver = new LogicalPlanMixinResolver();
+
+                @Override
+                public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+                  return MethodUtils.invokeMethod(resolver, method.getName(), args);
+                }
+              });
+
+      MethodUtils.invokeMethod(
+          objectMapper,
+          "registerModule",
+          Class.forName(UNSHADED_JACKSON_PACKAGE + ".module.scala.DefaultScalaModule$")
+              .getDeclaredField("MODULE$")
+              .get(null));
+
+      MethodUtils.invokeMethod(objectMapper, "setMixInResolver", resolver);
+    } catch (Exception | Error t) {
+      log.warn("Can't register jackson scala module for serializing LogicalPlan", t);
     }
+  }
 
-    mapper.setMixInResolver(new LogicalPlanMixinResolver());
+  private Object getObjectMapper() {
+    try {
+      return Class.forName(UNSHADED_JACKSON_PACKAGE + ".databind.ObjectMapper")
+          .getConstructor()
+          .newInstance();
+    } catch (InstantiationException
+        | InvocationTargetException
+        | NoSuchMethodException
+        | ClassNotFoundException
+        | IllegalAccessException e) {
```

Comment:
Do we really care about enumerating every single exception or are just ok with exception?